app/src/main/java/com/larryhsiao/nyx/android/jot/JotListFragment.java

```diff
@@ -0,0 +1,152 @@
+        }
+        if (jotIds.length != 0) {
+            return new QueriedJots(new JotsByIds(db, jotIds)).value();
+        } else {
+            return new QueriedJots(new AllJots(db)).value();
+        }
+    }
+
+    @Override
+    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.jot_list, menu);
+
+        SearchManager searchManager = ((SearchManager) getContext().getSystemService(Context.SEARCH_SERVICE));
+        MenuItem searchMenuItem = menu.findItem(R.id.menuItem_search);
+        SearchView searchView = (SearchView) searchMenuItem.getActionView();
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(getActivity().getComponentName()));
+        searchView.setOnCloseListener(() -> {
+            searchMenuItem.collapseActionView();
+            adapter.loadJots(loadJotsByArg());
+            return false;
+        });
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                adapter.loadJots(new QueriedJots(new JotsByKeyword(db, newText)).value());
+                return true;
+            }
+        });
+        searchMenuItem.setOnActionExpandListener(new MenuItem.OnActionExpandListener() {
+            @Override
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                return true;
+            }
+
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                searchView.setQuery("",true);
+                return true;
+            }
+        });
+        searchMenuItem.setOnMenuItemClickListener(item -> {
+            searchView.onActionViewExpanded();
+            return false;
+        });
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        if (item.getItemId() == R.id.menuItem_newJot) {
+            Fragment frag = new NewJotFragment();
+            frag.setTargetFragment(this, REQUEST_CODE_CREATE_JOT);
+            nextPage(frag);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @org.jetbrains.annotations.Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == REQUEST_CODE_CREATE_JOT) {
+            if (resultCode == RESULT_OK) {
+                adapter.insertJot(new JotById(new JotUriId(data.getData().toString()).value(), db).value());
+                getFragmentManager().popBackStack();
+            }
+        } else if (requestCode == REQUEST_CODE_JOT_CONTENT) {
+            if (resultCode == RESULT_OK) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5029215)